labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -29,39 +29,59 @@ object RecursionPatternMatchingExercise {
    * checkValuesIncrease(Seq(1,2,2)) == false
    */
   def checkValuesIncrease(seq: Seq[Int]): Boolean = {
-    error("fix me")
+    if (seq.length == 0) return false
+    if (seq.length == 1) return true
+    else
+      (seq(0) < seq(1)) match {
+      case true => return checkValuesIncrease(seq.tail)
+      case false => return false
+    }
   }
   
   /**
    * Group Consecutive values
    * List(1,1,2,3,1,1) -> List(1,1), List(2), List(3), List(1,1)
    */
-  def groupConsecutive[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+  def groupConsecutive[T](in: List[T]): List[List[T]] = in.length match{
+    case 0 => List()
+    case 1 => List(in)
+    case _ => subGroup(in) :: groupConsecutive(in.drop(subGroup(in).length))
+  }
+  
+  def subGroup[T](in: List[T]): List[T] = in.length match {
+    case 0 => List()
+    case 1 => in
+    case _ => in takeWhile {in.head ==}
```

Comment:
I wonder if this helper function is really required - could the takeWhile be better integrated into the main function?